api.c

```diff
@@ -572,10 +571,6 @@ int tcmu_emulate_evpd_inquiry(
 		/* Optimal xfer length */
 		memcpy(&data[12], &val32, 4);
 
-		/* MAXIMUM WRITE SAME LENGTH */
-		val64 = htobe64(VPD_MAX_WRITE_SAME_LENGTH);
-		memcpy(&data[36], &val64, 8);
-
```

Comment:
> Sorry for late, just came back from outside yesterday.

No problem. I will try to get your other patchset reviewed this week now too.

For linux, I was not referring to sg utils. I was referring to the kernel block/scsi support, so what you see in drivers/scsi/sd.c and in /sys/block/sdXYZ/queue and /sys/block/sdXYZ/device/scsi_disk.

> BTW, what does "report 0x3c for the b0 vpd page, but not actually support WS and so the MAX WS was zero in there" mean ? Is the 0x3c value for b0 vpd page's PAGE LENGTH field is incorrect ?

It is ok.

I was just speculating on why linux does not support the MAXIMUM WRITE SAME LENGTH = 0 meaning no limit like defined in the spec. Is it a bug in drivers/scsi/sd.c or was it intentional to try and support devices that misreported values. For the latter for example, before your patch to implement WS, we reported the MAX WS LENGTH as zero, but we did not support WS.